app/javascript/controllers/apex_nav_controller.js

```diff
@@ -1,81 +1,55 @@
 import { Controller } from "@hotwired/stimulus"
 
 export default class extends Controller {
-  static targets = ["startDate", "endDate", "interval", "uniques", "dropdays", "mainStart", "mainEnd", "tab"]
+  static targets = ["datePreset", "startDate", "endDate", "interval", "uniques", "dropdays", "tab", "card", "mainCard"]
 
-  updateDates(event) {
-    const [startDate, endDate] = JSON.parse(event.target.value)
-    this.mainStartTarget.value = startDate
-    this.mainEndTarget.value = endDate
-
-    this.mainStartTarget.focus()
-    this.mainStartTarget.blur()
-    this.mainEndTarget.focus()
-    this.mainEndTarget.blur()
-    event.target.focus()
-  }
-
-  updateStartDate(event) {
-    this.updateAllTargets(this.startDateTargets, event.target.value)
+  static values = {
+    mainCard: String,
+    agentsCard: String,
   }
 
-  updateEndDate(event) {
-    this.updateAllTargets(this.endDateTargets, event.target.value)
+  connect() {
+    this.tabTargets.forEach((tab) => {
+      if (tab.dataset.card === "main") {
+        if (tab.dataset.tab === this.mainCardValue) {
+          tab.click()
```

Comment:
this is my current solution for "removing" a default tab client-side. it's clearer in the view partial, but essentially i do not set any tab as active as a default, and this method basically just makes a check to see what the current `main_card` param is and auto-clicks the corresponding tab. this way, a user can bookmark when the view is on a different tab than Downloads and can load directly from there.

it's not quite the right solution because bootstrap is in the way, but it feels close.

Reply:
I think clicking a tab needs to alter the current query params.  Whether just manually changing via javascript, or actually firing a turbo request.

Then you can set the correct active tab in the ruby view.